src/main.py

```diff
@@ -0,0 +1,131 @@
+import preprocess.etl as etl
+import model.fitmodel as fitmodel
+import forecast.prediction as prediction
+
+import pathlib as pl
+import pandas as pd
+from sklearn.linear_model import LogisticRegression
+from sklearn.naive_bayes import GaussianNB
+from sklearn.svm import SVC
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.ensemble import RandomForestClassifier
+
+if __name__ == "__main__":
+    # step 1: ETL
+    # read in raw data
+    input_data_path = pl.Path(__file__).resolve().parents[1].joinpath('data/input/')  # the parent directory of
+    check_data = etl.Check(input_data_path)
+    df_all_files = check_data.list_all_files()
+    check_data.check_path(df_all_files)
+    check_data.convert(df_all_files)
+    df_raw = check_data.load(df_all_files)
+    check_data.check_size(df_raw)
+
```

Comment:
Your main file should only contains simple function calls, such as, load_data_and_check, transform_data, etc.

Reply:
Need suggestions on how to realize this: do I need to generate a "sub-main.py" for each of the module and then just call them in the main.py?